dor/providers/descriptor_generator.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+from datetime import datetime, UTC
+
+from dor.settings import S, template_env
+from dor.providers.models import PackageResource
+
+def relative_path_or_not(locref: str):
+    if locref.startswith("https://"):
+        return locref
+    return f"../{locref}"
+
+class DescriptorGenerator:
+    def __init__(self, package_path: Path, resources: list[PackageResource]):
+        self.package_path = package_path
+        self.resources = resources
+        self.entries = []
+
+    def write_files(self):
+        struct_map_locref_data = {}
+        for resource in self.resources:
+            if resource.type == "Asset":
+                identifier = f"urn:dor:{resource.id}"
+                struct_map_locref_data[identifier] = f"{resource.id}.{resource.type.lower()}.mets2.xml"
```

Comment:
Could make the file name generation a static method, since you're doing it twice.